src/cpu-kernels/sorting.cpp

```diff
@@ -575,14 +575,14 @@ ERROR awkward_IndexedArray_local_preparenext_64(
     int64_t* tocarry,
     const int64_t* starts,
     const int64_t* parents,
-    int64_t parentslength,
-    const int64_t* nextparents) {
+    const int64_t parentslength,
+    const int64_t* nextparents,
+    const int64_t nextlen) {
   int64_t j = 0;
   for (int64_t i = 0;  i < parentslength;  i++) {
     int64_t parent = parents[i];
     int64_t start = starts[parent];
-    int64_t nextparent = nextparents[j];
-    if (parent == nextparent) {
+    if (j < nextlen  &&  parent == nextparents[j]) {
```

Comment:
Oh, and this must be significant, too, right?

Reply:
That was it! Sometimes it could happen that `nextparent` at `nextlen` is equal to `parent`. For example, the changes in the `argsort` function from your branch triggered this problem in the `sort` function in a last test:
```
parents:
0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 
next parents:
0, 0, 0, 0, 1, 1, 2, 2, 2, 2, 0, 3, 5, 6, 1, 
starts:
0, 0, 0, 0, 0, 5, 5, 5, 5, 5, 10, 10, 10, 10, 10, 
tocarry:
0, 1, 2, 3, -1, 4, 5, -1, -1, -1, 6, 7, 8, 9, -1, 
```